clash-prelude/src/Clash/CPP.hs

```diff
@@ -11,7 +11,7 @@ Maintainer :  Christiaan Baaij <christiaan.baaij@gmail.com>
 #ifdef LARGE_TUPLES
 #define MAX_TUPLE_SIZE 62
 #else
-#define MAX_TUPLE_SIZE 12
+#define MAX_TUPLE_SIZE 15
```

Comment:
This is mostly here for our own comfort, as compilation gets _slow_ when increasing the tuple size, so I don't really want it to creep up and up.. would it be possible to change the tests, or is that too much trouble?

Reply:
So the `Generic` and `ShowX` instances went up to 15 in the previous code. I figured changing it here was the smallest change that 

  * kept things working that worked before (some tests need `ShowX` for more than 12-tuples)
  * removed some inaccuracy (instances for tuples larger than `MAX_TUPLE_SIZE`)

The alternative is to change about 6 or so tests, which isn't too bad but seems a bit odd for a PR which conceptually has nothing to do with the affected tests. I could go back and change the tests as part of #910, and then rebase onto that (lowering the limit back to 12)?